src/sqlancer/doris/DorisSchema.java

```diff
@@ -112,8 +112,25 @@ public int getSize() {
             return size;
         }
 
-        public static DorisCompositeDataType getRandomWithoutNull() {
+        public static DorisCompositeDataType getRandomWithoutNull(DorisGlobalState globalState) {
             DorisDataType type = DorisDataType.getRandomWithoutNull();
+            if (globalState != null) {
+                boolean typeIsQualified = false;
+                while (!typeIsQualified) {
+                    if (type == DorisDataType.INT && globalState.getDbmsSpecificOptions().testIntConstants
+                    || type == DorisDataType.BOOLEAN && globalState.getDbmsSpecificOptions().testBooleanConstants
+                    || type == DorisDataType.DECIMAL && globalState.getDbmsSpecificOptions().testDecimalConstants
+                    || type == DorisDataType.FLOAT && globalState.getDbmsSpecificOptions().testFloatConstants
+                    || type == DorisDataType.DATE && globalState.getDbmsSpecificOptions().testDateConstants
+                    || type == DorisDataType.DATETIME && globalState.getDbmsSpecificOptions().testDateTimeConstants
+                    || type == DorisDataType.VARCHAR && globalState.getDbmsSpecificOptions().testStringConstants
+                    ) {
+                        typeIsQualified = true;
```

Comment:
For example, the user does not want to test the datetime type, so the datetime type should not be generated from the `CREATE table`. I have corrected the writing style here.

Reply:
I think it would be better to select from a list of possible options, and throw an exception if no type can be selected. See https://github.com/sqlancer/sqlancer/blob/e3fbde0df29b386f100c371b1fcf11ff7005c7c6/src/sqlancer/duckdb/gen/DuckDBExpressionGenerator.java#L53 for this pattern.